cinema/tests/test_movie_api.py

```diff
@@ -0,0 +1,161 @@
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+
+def list_url():
+    return reverse("cinema:movie-list")
+
+
+def movie_session_list():
+    return reverse("cinema:moviesession-list")
+
+
+class MovieImageUploadTests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_superuser(
+            'admin', 'admin@myproject.com', 'password'
+        )
+        self.client.force_authenticate(self.user)
+        self.movie = sample_movie()
+        self.movie_session = sample_movie_session(movie=self.movie)
+
+    def tearDown(self):
+        self.movie.image.delete()
+
+    def test_upload_image_to_movie(self):
+        """Test uploading an image to movie"""
+        url = image_upload_url(self.movie.id)
+        with tempfile.NamedTemporaryFile(suffix='.jpg') as ntf:
+            img = Image.new('RGB', (10, 10))
+            img.save(ntf, format='JPEG')
+            ntf.seek(0)
+            res = self.client.post(url, {'image': ntf}, format='multipart')
+
+        self.movie.refresh_from_db()
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertIn('image', res.data)
+        self.assertTrue(os.path.exists(self.movie.image.path))
+
+    def test_upload_image_bad_request(self):
+        """Test uploading an invalid image"""
+        url = image_upload_url(self.movie.id)
+        res = self.client.post(url, {'image': 'not image'}, format='multipart')
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_post_image_to_movie_detail(self):
+        url = detail_url(self.movie.id)
+        with tempfile.NamedTemporaryFile(suffix='.jpg') as ntf:
+            img = Image.new('RGB', (10, 10))
+            img.save(ntf, format='JPEG')
+            ntf.seek(0)
+            res = self.client.post(
+                url,
+                {
+                    "title": "Title",
+                    "description": "Description",
+                    "duration": 90,
+                    'image': ntf,
+                },
+                format='multipart'
+            )
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Movie.objects.filter(title="Title").count(), 0)
+
+    def test_post_image_to_movie_list(self):
+        url = list_url()
+        with tempfile.NamedTemporaryFile(suffix='.jpg') as ntf:
+            img = Image.new('RGB', (10, 10))
+            img.save(ntf, format='JPEG')
+            ntf.seek(0)
+            res = self.client.post(
+                url,
+                {
+                    "title": "Title",
+                    "description": "Description",
+                    "duration": 90,
+                    'image': ntf,
+                },
+                format='multipart'
+            )
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(Movie.objects.filter(title="Title").count(), 0)
```

Comment:
https://stackoverflow.com/questions/2690521/django-check-for-any-exists-for-a-query

Reply:
Better check exists()